### PR TITLE
Fix bug in the burn time PR

### PR DESCRIPTION
**Pull request by Spinoscythe**
Fix the javadoc bug in #916

### CODE REVIEW

File: src/main/java/net/neoforged/neoforge/common/extensions/IItemStackExtension.java [hunk 1]
```diff
@@ -72,7 +72,7 @@ default boolean hasCraftingRemainingItem() {
     /**
      * Returns the fuel burn time for this item stack. If it is zero, this item is not a fuel.
      * <p>
-     * Will never return a negative value.
+     * Will never return a negative value. If so, the game will crash.
```
Comment (Technici4n) [inline src/main/java/net/neoforged/neoforge/common/extensions/IItemStackExtension.java]: Don't write that in the javadoc.

File: src/main/java/net/neoforged/neoforge/common/extensions/IItemExtension.java [hunk 2]
```diff
@@ -580,8 +580,7 @@ default boolean canDisableShield(ItemStack stack, ItemStack shield, LivingEntity
     }
 
     /**
-     * @return the fuel burn time for this item stack in a furnace. Return 0 to make
-     *         it not act as a fuel. Return -1 to let the default vanilla logic decide.
+     * @return the fuel burn time for this item stack in a furnace. Return 0 to make it not act as a fuel.
      * @apiNote This method takes precedence over the {@link net.neoforged.neoforge.registries.datamaps.builtin.NeoForgeDataMaps#FURNACE_FUELS data map}.
```
Comment (Technici4n) [inline src/main/java/net/neoforged/neoforge/common/extensions/IItemExtension.java:584]: this is also kind of incorrect given the default implementation

File: src/main/java/net/neoforged/neoforge/common/extensions/IItemExtension.java [hunk 3]
```diff
@@ -580,7 +580,7 @@ default boolean canDisableShield(ItemStack stack, ItemStack shield, LivingEntity
     }
 
     /**
-     * @return the fuel burn time for this item stack in a furnace. Return 0 to make it not act as a fuel.
+     * @return the fuel burn time for this item stack in a furnace. Returns 0 if value not present in the {@link net.neoforged.neoforge.registries.datamaps.builtin.NeoForgeDataMaps#FURNACE_FUELS burn time data map}
```
Comment (Technici4n) [inline src/main/java/net/neoforged/neoforge/common/extensions/IItemExtension.java:583]: This doesn't make sense?

Comment (Spinoscythe) [reply src/main/java/net/neoforged/neoforge/common/extensions/IItemExtension.java:583]: why what's wrong?

**Pull request closed without merging**